docs/docs/metrics.md

```diff
@@ -258,3 +258,126 @@ Dashes in metric names are replaced with underscores.
+as our alerting tool, but please keep in mind that any alerting service
+of your choice can be used instead.
+
+Please note that particular threshold values are dependent on many
+variables like the number of applications, a cluster size and so on.
+
+For the syntax of Prometheus alerting rules please refer to
+[its official documentation](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/).
+
+Alert whenever a Marathon instance goes down or restarts:
+
+```yaml
+  - alert: Marathon Uptime
+    expr: up{job="marathon"} == 0
+    for: 30s
+    labels:
+      severity: warning
+    annotations:
+      summary: It fires when a Marathon instance stops or restarts.
+```
+
+Alert when there are less than 2 out of 3 Marathon instances that are
+up and running:
+
+```yaml
+  - alert: Marathon Too Few Instances
+    expr: sum(up{job="marathon"}) < 2
+    for: 30s
+    labels:
+      severity: critical
+      annotations:
+      summary: It fires when there are fewer Marathon instances up and running than expected.
+```
+
+Alert if Marathon's heap usage is too high:
+
+```yaml
+  - alert: Marathon Heap Usage
+    expr: avg_over_time(marathon_jvm_memory_heap_used_gauge_bytes[10s]) > (5 * 1024 * 1024 * 1024)
+    for: 10s
+    labels:
+      severity: critical
+    annotations:
+      summary: It fires when Marathon's heap is too big.
+```
+
+Alert if Marathon has too many threads:
+
+```yaml
+  - alert: Marathon Threads
+    expr: avg_over_time(marathon_jvm_threads_active_gauge[10s]) > 300
+    for: 10s
+    labels:
+      severity: critical
+    annotations:
+      summary: It fires when Marathon has too many threads.
+```
+
+Alert if Marathon dismisses deployments way too often:
+
+```yaml
+  - alert: Marathon Dismissed Deployments
+    expr: rate(marathon_deployments_dismissed_counter[1m]) > 3
+    for: 1m
+    labels:
+      severity: warning
+    annotations:
+      summary: It fires if there are too many deployments dismissed by Marathon.
+```
+
+Alert if Marathon receives offers, but is unable to use them for some
+reason, i.e. insufficient resources:
+
+```yaml
+  - alert: Marathon Given No Suitable Offer
+    expr: rate(marathon_mesos_offers_used_counter[1m]) == 0 and rate(marathon_mesos_offers_incoming_counter[1m]) > 0
+    for: 1m
+    labels:
+      severity: warning
+    annotations:
+      summary: It fires if there are incoming Mesos offers, but none can be used by Marathon.
+```
+
+Alert if there are too many concurrent HTTP requests to Marathon:
+
+```yaml
+  - alert: Marathon HTTP Requests
+    expr: avg_over_time(marathon_http_requests_active_gauge[10s]) > 100
+    for: 10s
+    labels:
+      severity: critical
+    annotations:
+      summary: It fires if there are too many concurrent HTTP requests to Marathon.
+```
+
+Alert if Marathon responds with a 5xx status code:
+
+```yaml
+  - alert: Marathon HTTP 5xx Responses
+    expr: marathon_http_responses_5xx_rate_m1_rate > 0
```

Comment:
Since `5xx` responses are expected during e.g. leader reconciliation, I'd rather go for `marathon_http_responses_api_error_rate_m1_rate`...which is only implemented in the deprecated [metrics](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/HttpResponseMetricsHandler.scala#L20)? I'm in favor of adding it to the new metrics and using it here.

Reply:
The deprecated metric hides real failures. It ignores HTTP 503, which is about something else, not a bad gateway. So HTTP 502 should be used instead, but Marathon returns 503, if it is unable to forward a request. It is potentially a breaking change, though I am all pro it. What do you think? I would do it in a new PR though.

Reply:
if different http 5XX can mean different things for alerting, shouldn't we maybe monitor them separately as well, not just inside 5xx metric?

But to your original thought - that sounds reasonable. If error metric is not actually monitoring errors I would consider it a bugfix :) So I guess - go for it 👍

Reply:
I support changing our exit codes to match the http spec. I think it’s unlikely that clients are hard coded to care about anything except it’s a 5xx

Reply:
I think it is fine to include 5xx responses during leader election. I’d Leave as is.

Reply:
@timcharper, do I understand you correctly

1. you are pro returning 502 instead of 503 when request forwarding fails
2. there is no need to introduce a new metric similar to the old one which ignores 503 (or 502) status codes, in addition to the existing ones?

?

Reply:
Hmm, I wasn’t aware that we hid leader forwarding 5xx responses them previously. I would not hide them.

I support using 502 to indicate proxy failures. Also... 504 if proxy times out?

Reply:
504 is already sent if it times out:
```
            case Failure(ex: SocketTimeoutException) =>
              logger.error(ERROR_STATUS_GATEWAY_TIMEOUT, ex)
              response.sendError(GatewayTimeout.intValue, ERROR_STATUS_GATEWAY_TIMEOUT)
```